jsurfer-core/src/main/java/org/jsfr/json/compiler/JsonPathCompiler.java

```diff
@@ -218,6 +215,26 @@ public Void visitFilterNEqualBool(JsonPathParser.FilterNEqualBoolContext ctx) {
         return rst;
     }
 
+    @Override
+    public Void visitFilterEqualNull(JsonPathParser.FilterEqualNullContext ctx) {
+        filterPathBuilder = createFilterPathBuilder();
+        Void rst = super.visitFilterEqualNull(ctx);
+        filterBuilder.append(
+            new EqualityNullPredicate(filterPathBuilder.build()));
+        filterPathBuilder = null;
```

Comment:
Why do you set local variable to null?

Reply:
I think `super.visitFilterEqualNull` indirectly modifies contents of `filterPathBuilder` once called (in this case anyway) and once we're done with the construction of the filter object, we discard "temporary" variables right away for GC (and to avoid potential processing errors i.e. when some method is called that checks if filterPathBuilder is not null and should be included in the processing step), in this case `filterPathBuilder`. 

Overall from my limited understanding the Compiler is stateful and it just walks from left to right over every single operator in the expression statefully until its out of tokens to process.  So if the expression has multiple filters in it, it would be constructing and discarding `filterPathBuilder` multiple times over the course of compilation since its just a "temporary" variable that is shared between methods.